tests/util/ConverterTestCase.py

```diff
@@ -23,9 +23,33 @@ class ConverterTestCase(usdex.test.TestCase):
         omni.asset_validator.IssuePredicates.ContainsMessage("ArticulationRootAPI definition on a kinematic rigid body is not allowed"),
     ]
 
-    def assert_rotation_almost_equal(self, rot1: Gf.Rotation, rot2: Gf.Rotation, tolerance: float = 1e-6):
-        self.assertTrue(Gf.IsClose(rot1.GetAxis(), rot2.GetAxis(), tolerance), f"Axis mismatch: {rot1.GetAxis()} != {rot2.GetAxis()}")
-        self.assertTrue(Gf.IsClose(rot1.GetAngle(), rot2.GetAngle(), tolerance), f"Angle mismatch: {rot1.GetAngle()} != {rot2.GetAngle()}")
+    def assert_rotation_almost_equal(self, rot1: Gf.Rotation | Gf.Quatf | Gf.Quatd, rot2: Gf.Rotation | Gf.Quatf | Gf.Quatd, tolerance: float = 1e-6):
+        """Assert two rotations are almost equal (same concrete type required).
+        Strict: Rotations must be the same class (Rotation vs Quatf vs Quatd). Antipodal quaternions
+        are treated as different.
+        """
+        # Enforce identical concrete type without direct type comparison triggering E721.
+        if not (isinstance(rot1, type(rot2)) and isinstance(rot2, type(rot1))):
+            self.fail(f"Rotation types do not match or are unsupported: {type(rot1)} vs {type(rot2)}")
```

Comment:
Why can't we gracefully compare across types? A loose tolerance _should_ allow `Gf.Quatf ~= Gf.Quatd` if that's what someone wants to assert

Or maybe we don't want to do that to avoid accidental return type differences?